pkg/engine/update_test.go

```diff
@@ -49,3 +54,42 @@ func TestAbbreviateFilePath(t *testing.T) {
 		assert.Equal(t, filepath.ToSlash(tt.expected), filepath.ToSlash(actual))
 	}
 }
+
+func TestDeletingCustomResourceProducesResourceOutputsEvent(t *testing.T) {
+	t.Parallel()
+
+	cancelCtx, _ := cancel.NewContext(context.Background())
+
+	acts := newUpdateActions(&Context{
+		Cancel: cancelCtx,
+	}, nil, deploymentOptions{})
+	eventsChan := make(chan Event, 10)
+	acts.Opts.Events.ch = eventsChan
+
+	step := deploy.NewDeleteStep(&deploy.Deployment{}, map[resource.URN]bool{}, &resource.State{
+		URN:      resource.URN("urn:pulumi:stack::project::my:example:Foo::foo"),
+		ID:       "foo",
+		Custom:   true,
```

Comment:
Does this test pass without the product change? Shouldn't this be `false` to test component resources?

```suggestion
		Custom:   false,
```